lib/bolt/catalog.rb

```diff
@@ -92,6 +93,16 @@ def generate_ast(code)
       end
     end
 
+    def validate_hiera_config(hiera_config)
+      if File.exist?(File.path(hiera_config))
+        data = Bolt::Util.read_config_file(hiera_config, nil, 'hiera_config')
```

Comment:
This is overly specific. `read_config_file` has a specific use case: at takes an explicitly specified path (that may be nil), and a list of fallbacks. If the specified path is not nil, then it will error if not found; for the fallbacks, it'll only log if not found.

This could simply be `data = File.open(path, "r:UTF-8") { |f| YAML.safe_load(f.read) }`.

We should also consider where and whether we should error if `hiera_config` is explicitly configured in bolt.yaml but not found.

Reply:
I guess on the plus side, `read_config_file` gives us consistent logging.